### PR TITLE
[expo-image] [ENG-10268] support tintColor for SVGs on Android

### PR DESCRIPTION
**Pull request by kadikraman**
# Why

This is actually a new feature. We've not supported `tintColor` on SVGs before, but iOS support was added [recently](https://github.com/expo/expo/pull/23418). This will add Android support. 

# How

SVG on Android needs to be converted into a format that supports `tintColor`.

# Test Plan

Test on Bare Expo.

| Before      | After |
| ----------- | ----------- |
| Screenshot 2023-10-04 at 15 28 43 | Screenshot 2023-10-04 at 15 56 01 |

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by alanjhughes — APPROVED** [review]

**Comment by Saadnajmi** [comment]
:o TIL expo-image supports SVGs, like `react-native-svg`!

**Comment by kadikraman** [comment]
@Saadnajmi quite so, though with a decent amount of faff 😂 - you need a custom decoder to make SVGs work with [Glide](https://github.com/bumptech/glide), and it's not recommended to use a ton of them on Android (for performance).

**Comment by tsapeta** [comment]
I'm wondering why @lukmccall and me were not automatically requested for a review as we're code owners of expo-image. @lukmccall has already spent some time to investigate this issue, he definitely needs to take a look if this solution is actually correct (doesn't break other things).

@Saadnajmi It does, but imho they have a bit different purposes. `react-native-svg` is still a better option if you want to draw some parts of the SVG dynamically, animate them and so on. If you just need to render a static SVG, `expo-image` seems to be a better fit, also due to performance reasons.

**Comment by kadikraman** [comment]
Yes for sure @tsapeta @lukmccall please take a look. This change is merged but not published. You are defintiely [codeowners](https://github.com/expo/expo/blob/a148f727996df02c3c489cccd6c17ce1c6b18afa/.github/CODEOWNERS#L40) it so it should have requested a review from you 🤔

**Comment by lukmccall** [comment]
When I was implementing the same future, I did the same thing. However, we didn't merge it because that code adds some side effects. By converting the picture to drawable, you changed how that asset will be drawn on the screen. There are a couple of differences. In summary, Drawables are primarily used for managing static images and icons that are loaded from resources and displayed in UI elements, while Picture is a more low-level graphics container used for programmatic drawing and custom graphics rendering. They serve different purposes and are used in different contexts within Android development.

So when you declare in SVG `width` or `heigh`, your assets will be drawn on a canvas of that size and later resized if the native view is bigger. In that case, you might end up with a pixilated image. 
image
I'm not entirely sure if that's the correct behavior or not. I have mixed feelings about it. On one hand, it's what the browser would normally do, which makes it something users might expect. On the other hand, most users would assume that if they use an SVG image, it should be able to scale infinitely.

By default, the SVG has a view size of 512 by 512. If you convert the picture to a drawable, it will use those dimensions which can waste memory and CPU cycles if you want to draw something smaller.

If you want to change the size of the SVG dynamically, let's say you will resize the Image view; it won't scale correctly. That will be done using interpolation with static images rather than scaling the vector graphic. 

Perhaps we should only use your flow if the tint color is defined, otherwise, leave the Picture class as it is. What are your thoughts are.

**Comment by kadikraman** [comment]
@lukmccall thank you for your comments! You're right there a trade-off here. The problem is the `width` and `height` passed in [here](https://github.com/expo/expo/blob/f38bb6d6b9bab10879f1e4e1cfef9a5062366768/packages/expo-image/android/src/main/java/expo/modules/image/svg/SVGDecoder.kt) in the svg decoder.

For an SVG that is loaded via a url, the `width` and `height` are actually passed in correctly. So e.g. the React logo rendered in Bare Expo would get `width=100*pixelDensity` and `height=100*pixelDensity` which is exactly what we want even though the svg source does not list these values at all. So Glide must be somehow adding the metadata here?

This is what the source for the React SVG looks like, it definitely doesn't have any width/height data:
```svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="-11.5 -10.23174 23 20.46348">
  <title>React Logo</title>
  <circle cx="0" cy="0" r="2.05" fill="#61dafb"/>
  <g stroke="#61dafb" stroke-width="1" fill="none">
    <ellipse rx="11" ry="4.2"/>
    <ellipse rx="11" ry="4.2" transform="rotate(60)"/>
    <ellipse rx="11" ry="4.2" transform="rotate(120)"/>
  </g>
</svg>

```

But if you use a _local_ svg it uses the `viewBox` width and height. However if you set the width and height on the svg itself then they are used instead! (exact values though, they don't account for pixel density).

If there was a way to pass in the correct display size of the `ExpoImage` in `SVGDecoder.kt` and use that to size the svg, that would resolve this issue completely, can you see if that's in any way possible to do?

**Comment by lukmccall** [comment]
> For an SVG that is loaded via a url, the width and height are actually passed in correctly. So e.g. the React logo rendered in Bare Expo would get width=100*pixelDensity and height=100*pixelDensity which is exactly what we want even though the svg source does not list these values at all. So Glide must be somehow adding the metadata here?

It doesn't add any metadata. It knows the size of the view that is going to display the image, so it can resize that correctly. 

> If there was a way to pass in the correct display size of the ExpoImage in SVGDecoder.kt and use that to size the svg, that would resolve this issue completely, can you see if that's in any way possible to do?

We can try to do it. The size is calculated here https://github.com/expo/expo/blob/0fa59a4f85aa4c7b0e942f05afb4cc03c6d2e7cd/packages/expo-image/android/src/main/java/expo/modules/image/ImageViewWrapperTarget.kt#L97-L106. However, that won't solve olde the issues. Right now, the SVG will resize automatically and be pixel-perfect scaling. When converting to drawable,  the SVG has to be decoded multiple times.

**Pull request merged**
